pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -514,15 +592,22 @@ func (nc NodeController) tryUpdateNodeStatus(node *api.Node) (error, time.Durati
 				nc.recordNodeOfflineEvent(node)
 			}
 		}
-		_, err = nc.kubeClient.Nodes().Update(node)
-		if err != nil {
-			glog.Errorf("Error updating node %s: %v", node.Name, err)
-		} else {
-			return nil, gracePeriod, lastReadyCondition
+		if !api.Semantic.DeepEqual(nc.getCondition(&node.Status, api.NodeReady), lastReadyCondition) {
+			if _, err = nc.kubeClient.Nodes().Update(node); err != nil {
+				glog.Errorf("Error updating node %s: %v", node.Name, err)
+				return gracePeriod, lastReadyCondition, readyCondition, err
+			} else {
+				nc.nodeStatusMap[node.Name] = NodeStatusData{
+					status:                   node.Status,
+					probeTimestamp:           nc.nodeStatusMap[node.Name].probeTimestamp,
```

Comment:
just to confirm, this == savedNodeStatus.probeTimestamp, right? (no need to change, just want to make sure I'm not missing omething)